controllers/resource/template.go

```diff
@@ -105,7 +105,7 @@ func (r *VsphereTemplate) TemplateResources(ctx context.Context, eksaCluster *an
 		if err != nil {
 			return nil, err
 		}
-		if vsphere.NeedsNewKubeadmConfigTemplate(&workerNodeGroupConfiguration, oldWn, oldVmc, &vmc) {
+		if needsVSphereNewKubeadmConfigTemplate(&workerNodeGroupConfiguration, oldWn, oldVmc, &vmc) {
```

Comment:
using a different algorithm to compare SSH keys
Since we try to rebuild the vsphere machine config from the vsphere maachine template by reversing the implicit conversion logic of the template builder, we need to be smart when comparing these. Such conversion logic is a one way street where data is lost, so a perfect reverse conversion is impossible. In particular, the `user@host` part is lost
This new method takes that into account